### PR TITLE
Assorted cleanup commits

### PR DESCRIPTION
**Pull request by digitalinfinity**
1) Cleanup some CMakeList filesl
2) Fix unnecessary reference to profiler thunk
3) Add simple test runner for hello world so that CI can run it

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -96,6 +97,20 @@ if(CLR_CMAKE_PLATFORM_UNIX)
...
+
+    add_compile_options(
+        -fms-extensions
+        -msse4.1
+        "$<$<COMPILE_LANGUAGE:C>:${c_warn_flags}>"
+        "$<$<COMPILE_LANGUAGE:CXX>:${cxx_warn_flags}>"
+        "$<$<COMPILE_LANGUAGE:CXX>:${cxx_compile_flags}>"
+    )
+
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdelayed-template-parsing")
```
Comment (jianchun) [inline CMakeLists.txt]: `-fdelayed-template-parsing` also specified in `cxx_compile_flags/add_compile_options`?

File: lib/Runtime/CMakeLists.txt [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-    $(CHAKRA_RUNTIME_SOURCE_DIR)/ByteCode
-    $(CHAKRA_RUNTIME_SOURCE_DIR)/PlatformAgnostic
-    $(CHAKRA_RUNTIME_SOURCE_DIR)/Math
+    .
+    ../Common
+    ../Backend
+    ../Parser
+    ./ByteCode
+    ./PlatformAgnostic
+    ./Math
```
Comment (jianchun) [inline lib/Runtime/CMakeLists.txt]: These can omit `./`

Comment (jianchun) [inline lib/Runtime/CMakeLists.txt]: LGTM

File: CMakeLists.txt (hunk 1 above)
Comment (digitalinfinity) [inline CMakeLists.txt]: Gah- meant to remove it from cmake_cxx_flags- removed.

File: lib/Runtime/CMakeLists.txt (hunk 2 above)
Comment (digitalinfinity) [inline lib/Runtime/CMakeLists.txt]: Done- thanks for the review!

**Pull request merged**